exonum/src/blockchain/config.rs

```diff
@@ -295,6 +296,131 @@ impl ValidateInput for ConsensusConfig {
     }
 }
 
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    ProtobufConvert,
+    BinaryValue,
+    ObjectHash,
+)]
+#[protobuf_convert(source = "runtime::GenesisConfig")]
+pub struct GenesisConfig {
+    /// Blockchain configuration used to create the genesis block.
+    pub consensus_config: ConsensusConfig,
+
+    /// Artifacts specification of builtin services.
+    pub artifacts: Vec<ArtifactSpec>, // TODO: HashSet? Duplications? Perhaps check in builder?
```

Comment:
I think `Vec` is ok.

Regarding checks: you can impement a `validate` method similar to one in `consensus_config` which will check that (and call `consensus_config.validate()` as well).